README.md
# client

The Jira Cloud platform REST API

- API version: 1001.0.0-SNAPSHOT

- Build date: 2020-12-31T17:44:39.790417-08:00[America/Los_Angeles]

Jira Cloud platform REST API documentation


*Automatically generated by the [OpenAPI Generator](https://openapi-generator.tech)*

## Requirements

Building the API client library requires:

1. Java 1.8+
2. Maven/Gradle

## Installation

To install the API client library to your local Maven repository, simply execute:

```shell
mvn clean install
```

To deploy it to a remote Maven repository instead, configure the settings of the repository and execute:

```shell
mvn clean deploy
```

Refer to the [OSSRH Guide](http://central.sonatype.org/pages/ossrh-guide.html) for more information.

### Maven users

Add this dependency to your project's POM:

```xml
<dependency>
  <groupId>com.atlassian.jira.rest</groupId>
  <artifactId>client</artifactId>
  <version>0.0.1-SNAPSHOT</version>
  <scope>compile</scope>
</dependency>
```

### Gradle users

Add this dependency to your project's build file:

```groovy
compile "com.atlassian.jira.rest:client:0.0.1-SNAPSHOT"
```

### Others

At first generate the JAR by executing:

```shell
mvn clean package
```

Then manually install the following JARs:

- `target/client-0.0.1-SNAPSHOT.jar`
- `target/lib/*.jar`

## Getting Started

Please follow the [installation](#installation) instruction and execute the following Java code:

```java

import com.atlassian.jira.rest.client.*;
import com.atlassian.jira.rest.client.auth.*;
import com.atlassian.jira.rest.client.model.*;
import com.atlassian.jira.rest.client.api.AppPropertiesApi;

public class AppPropertiesApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();
        defaultClient.setBasePath("https://your-domain.atlassian.com");
        
        AppPropertiesApi apiInstance = new AppPropertiesApi(defaultClient);
        String addonKey = "addonKey_example"; // String | The key of the app, as defined in its descriptor.
        String propertyKey = "propertyKey_example"; // String | The key of the property.
        try {
            apiInstance.addonPropertiesResourceDeleteAddonPropertyDelete(addonKey, propertyKey);
        } catch (ApiException e) {
            System.err.println("Exception when calling AppPropertiesApi#addonPropertiesResourceDeleteAddonPropertyDelete");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

```

## Documentation for API Endpoints

All URIs are relative to *https://your-domain.atlassian.com*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*AppPropertiesApi* | [**addonPropertiesResourceDeleteAddonPropertyDelete**](docs/AppPropertiesApi.md#addonPropertiesResourceDeleteAddonPropertyDelete) | **DELETE** /rest/atlassian-connect/1/addons/{addonKey}/properties/{propertyKey} | Delete app property
*AppPropertiesApi* | [**addonPropertiesResourceGetAddonPropertiesGet**](docs/AppPropertiesApi.md#addonPropertiesResourceGetAddonPropertiesGet) | **GET** /rest/atlassian-connect/1/addons/{addonKey}/properties | Get app properties
*AppPropertiesApi* | [**addonPropertiesResourceGetAddonPropertyGet**](docs/AppPropertiesApi.md#addonPropertiesResourceGetAddonPropertyGet) | **GET** /rest/atlassian-connect/1/addons/{addonKey}/properties/{propertyKey} | Get app property
*AppPropertiesApi* | [**addonPropertiesResourcePutAddonPropertyPut**](docs/AppPropertiesApi.md#addonPropertiesResourcePutAddonPropertyPut) | **PUT** /rest/atlassian-connect/1/addons/{addonKey}/properties/{propertyKey} | Set app property
*ApplicationRolesApi* | [**getAllApplicationRoles**](docs/ApplicationRolesApi.md#getAllApplicationRoles) | **GET** /rest/api/3/applicationrole | Get all application roles
*ApplicationRolesApi* | [**getApplicationRole**](docs/ApplicationRolesApi.md#getApplicationRole) | **GET** /rest/api/3/applicationrole/{key} | Get application role
*AuditRecordsApi* | [**getAuditRecords**](docs/AuditRecordsApi.md#getAuditRecords) | **GET** /rest/api/3/auditing/record | Get audit records
*AvatarsApi* | [**deleteAvatar**](docs/AvatarsApi.md#deleteAvatar) | **DELETE** /rest/api/3/universal_avatar/type/{type}/owner/{owningObjectId}/avatar/{id} | Delete avatar
*AvatarsApi* | [**getAllSystemAvatars**](docs/AvatarsApi.md#getAllSystemAvatars) | **GET** /rest/api/3/avatar/{type}/system | Get system avatars by type
*AvatarsApi* | [**getAvatars**](docs/AvatarsApi.md#getAvatars) | **GET** /rest/api/3/universal_avatar/type/{type}/owner/{entityId} | Get avatars
*AvatarsApi* | [**storeAvatar**](docs/AvatarsApi.md#storeAvatar) | **POST** /rest/api/3/universal_avatar/type/{type}/owner/{entityId} | Load avatar
*DashboardsApi* | [**copyDashboard**](docs/DashboardsApi.md#copyDashboard) | **POST** /rest/api/3/dashboard/{id}/copy | Copy dashboard
*DashboardsApi* | [**createDashboard**](docs/DashboardsApi.md#createDashboard) | **POST** /rest/api/3/dashboard | Create dashboard
*DashboardsApi* | [**deleteDashboard**](docs/DashboardsApi.md#deleteDashboard) | **DELETE** /rest/api/3/dashboard/{id} | Delete dashboard
*DashboardsApi* | [**deleteDashboardItemProperty**](docs/DashboardsApi.md#deleteDashboardItemProperty) | **DELETE** /rest/api/3/dashboard/{dashboardId}/items/{itemId}/properties/{propertyKey} | Delete dashboard item property
*DashboardsApi* | [**getAllDashboards**](docs/DashboardsApi.md#getAllDashboards) | **GET** /rest/api/3/dashboard | Get all dashboards
*DashboardsApi* | [**getDashboard**](docs/DashboardsApi.md#getDashboard) | **GET** /rest/api/3/dashboard/{id} | Get dashboard
*DashboardsApi* | [**getDashboardItemProperty**](docs/DashboardsApi.md#getDashboardItemProperty) | **GET** /rest/api/3/dashboard/{dashboardId}/items/{itemId}/properties/{propertyKey} | Get dashboard item property
*DashboardsApi* | [**getDashboardItemPropertyKeys**](docs/DashboardsApi.md#getDashboardItemPropertyKeys) | **GET** /rest/api/3/dashboard/{dashboardId}/items/{itemId}/properties | Get dashboard item property keys
*DashboardsApi* | [**getDashboardsPaginated**](docs/DashboardsApi.md#getDashboardsPaginated) | **GET** /rest/api/3/dashboard/search | Search for dashboards
*DashboardsApi* | [**setDashboardItemProperty**](docs/DashboardsApi.md#setDashboardItemProperty) | **PUT** /rest/api/3/dashboard/{dashboardId}/items/{itemId}/properties/{propertyKey} | Set dashboard item property
*DashboardsApi* | [**updateDashboard**](docs/DashboardsApi.md#updateDashboard) | **PUT** /rest/api/3/dashboard/{id} | Update dashboard
*DynamicModulesApi* | [**dynamicModulesResourceGetModulesGet**](docs/DynamicModulesApi.md#dynamicModulesResourceGetModulesGet) | **GET** /rest/atlassian-connect/1/app/module/dynamic | Get modules
*DynamicModulesApi* | [**dynamicModulesResourceRegisterModulesPost**](docs/DynamicModulesApi.md#dynamicModulesResourceRegisterModulesPost) | **POST** /rest/atlassian-connect/1/app/module/dynamic | Register modules
*DynamicModulesApi* | [**dynamicModulesResourceRemoveModulesDelete**](docs/DynamicModulesApi.md#dynamicModulesResourceRemoveModulesDelete) | **DELETE** /rest/atlassian-connect/1/app/module/dynamic | Remove modules
*FilterSharingApi* | [**addSharePermission**](docs/FilterSharingApi.md#addSharePermission) | **POST** /rest/api/3/filter/{id}/permission | Add share permission
*FilterSharingApi* | [**deleteSharePermission**](docs/FilterSharingApi.md#deleteSharePermission) | **DELETE** /rest/api/3/filter/{id}/permission/{permissionId} | Delete share permission
*FilterSharingApi* | [**getDefaultShareScope**](docs/FilterSharingApi.md#getDefaultShareScope) | **GET** /rest/api/3/filter/defaultShareScope | Get default share scope
*FilterSharingApi* | [**getSharePermission**](docs/FilterSharingApi.md#getSharePermission) | **GET** /rest/api/3/filter/{id}/permission/{permissionId} | Get share permission
*FilterSharingApi* | [**getSharePermissions**](docs/FilterSharingApi.md#getSharePermissions) | **GET** /rest/api/3/filter/{id}/permission | Get share permissions
*FilterSharingApi* | [**setDefaultShareScope**](docs/FilterSharingApi.md#setDefaultShareScope) | **PUT** /rest/api/3/filter/defaultShareScope | Set default share scope
*FiltersApi* | [**createFilter**](docs/FiltersApi.md#createFilter) | **POST** /rest/api/3/filter | Create filter
*FiltersApi* | [**deleteFavouriteForFilter**](docs/FiltersApi.md#deleteFavouriteForFilter) | **DELETE** /rest/api/3/filter/{id}/favourite | Remove filter as favorite
*FiltersApi* | [**deleteFilter**](docs/FiltersApi.md#deleteFilter) | **DELETE** /rest/api/3/filter/{id} | Delete filter
*FiltersApi* | [**getColumns**](docs/FiltersApi.md#getColumns) | **GET** /rest/api/3/filter/{id}/columns | Get columns
*FiltersApi* | [**getFavouriteFilters**](docs/FiltersApi.md#getFavouriteFilters) | **GET** /rest/api/3/filter/favourite | Get favorite filters
*FiltersApi* | [**getFilter**](docs/FiltersApi.md#getFilter) | **GET** /rest/api/3/filter/{id} | Get filter
*FiltersApi* | [**getFilters**](docs/FiltersApi.md#getFilters) | **GET** /rest/api/3/filter | Get filters
*FiltersApi* | [**getFiltersPaginated**](docs/FiltersApi.md#getFiltersPaginated) | **GET** /rest/api/3/filter/search | Search for filters
*FiltersApi* | [**getMyFilters**](docs/FiltersApi.md#getMyFilters) | **GET** /rest/api/3/filter/my | Get my filters
*FiltersApi* | [**resetColumns**](docs/FiltersApi.md#resetColumns) | **DELETE** /rest/api/3/filter/{id}/columns | Reset columns
*FiltersApi* | [**setColumns**](docs/FiltersApi.md#setColumns) | **PUT** /rest/api/3/filter/{id}/columns | Set columns
*FiltersApi* | [**setFavouriteForFilter**](docs/FiltersApi.md#setFavouriteForFilter) | **PUT** /rest/api/3/filter/{id}/favourite | Add filter as favorite
*FiltersApi* | [**updateFilter**](docs/FiltersApi.md#updateFilter) | **PUT** /rest/api/3/filter/{id} | Update filter
*GroupAndUserPickerApi* | [**findUsersAndGroups**](docs/GroupAndUserPickerApi.md#findUsersAndGroups) | **GET** /rest/api/3/groupuserpicker | Find users and groups
*GroupsApi* | [**addUserToGroup**](docs/GroupsApi.md#addUserToGroup) | **POST** /rest/api/3/group/user | Add user to group
*GroupsApi* | [**bulkGetGroups**](docs/GroupsApi.md#bulkGetGroups) | **GET** /rest/api/3/group/bulk | Bulk get groups
*GroupsApi* | [**createGroup**](docs/GroupsApi.md#createGroup) | **POST** /rest/api/3/group | Create group
*GroupsApi* | [**findGroups**](docs/GroupsApi.md#findGroups) | **GET** /rest/api/3/groups/picker | Find groups
*GroupsApi* | [**getGroup**](docs/GroupsApi.md#getGroup) | **GET** /rest/api/3/group | Get group
*GroupsApi* | [**getUsersFromGroup**](docs/GroupsApi.md#getUsersFromGroup) | **GET** /rest/api/3/group/member | Get users from group
*GroupsApi* | [**removeGroup**](docs/GroupsApi.md#removeGroup) | **DELETE** /rest/api/3/group | Remove group
*GroupsApi* | [**removeUserFromGroup**](docs/GroupsApi.md#removeUserFromGroup) | **DELETE** /rest/api/3/group/user | Remove user from group
*IssueAttachmentsApi* | [**addAttachment**](docs/IssueAttachmentsApi.md#addAttachment) | **POST** /rest/api/3/issue/{issueIdOrKey}/attachments | Add attachment
*IssueAttachmentsApi* | [**expandAttachmentForHumans**](docs/IssueAttachmentsApi.md#expandAttachmentForHumans) | **GET** /rest/api/3/attachment/{id}/expand/human | Get all metadata for an expanded attachment
*IssueAttachmentsApi* | [**expandAttachmentForMachines**](docs/IssueAttachmentsApi.md#expandAttachmentForMachines) | **GET** /rest/api/3/attachment/{id}/expand/raw | Get contents metadata for an expanded attachment
*IssueAttachmentsApi* | [**getAttachment**](docs/IssueAttachmentsApi.md#getAttachment) | **GET** /rest/api/3/attachment/{id} | Get attachment metadata
*IssueAttachmentsApi* | [**getAttachmentMeta**](docs/IssueAttachmentsApi.md#getAttachmentMeta) | **GET** /rest/api/3/attachment/meta | Get Jira attachment settings
*IssueAttachmentsApi* | [**removeAttachment**](docs/IssueAttachmentsApi.md#removeAttachment) | **DELETE** /rest/api/3/attachment/{id} | Delete attachment
*IssueCommentPropertiesApi* | [**deleteCommentProperty**](docs/IssueCommentPropertiesApi.md#deleteCommentProperty) | **DELETE** /rest/api/3/comment/{commentId}/properties/{propertyKey} | Delete comment property
*IssueCommentPropertiesApi* | [**getCommentProperty**](docs/IssueCommentPropertiesApi.md#getCommentProperty) | **GET** /rest/api/3/comment/{commentId}/properties/{propertyKey} | Get comment property
*IssueCommentPropertiesApi* | [**getCommentPropertyKeys**](docs/IssueCommentPropertiesApi.md#getCommentPropertyKeys) | **GET** /rest/api/3/comment/{commentId}/properties | Get comment property keys
*IssueCommentPropertiesApi* | [**setCommentProperty**](docs/IssueCommentPropertiesApi.md#setCommentProperty) | **PUT** /rest/api/3/comment/{commentId}/properties/{propertyKey} | Set comment property
*IssueCommentsApi* | [**addComment**](docs/IssueCommentsApi.md#addComment) | **POST** /rest/api/3/issue/{issueIdOrKey}/comment | Add comment
*IssueCommentsApi* | [**deleteComment**](docs/IssueCommentsApi.md#deleteComment) | **DELETE** /rest/api/3/issue/{issueIdOrKey}/comment/{id} | Delete comment
*IssueCommentsApi* | [**getComment**](docs/IssueCommentsApi.md#getComment) | **GET** /rest/api/3/issue/{issueIdOrKey}/comment/{id} | Get comment
*IssueCommentsApi* | [**getComments**](docs/IssueCommentsApi.md#getComments) | **GET** /rest/api/3/issue/{issueIdOrKey}/comment | Get comments
*IssueCommentsApi* | [**getCommentsByIds**](docs/IssueCommentsApi.md#getCommentsByIds) | **POST** /rest/api/3/comment/list | Get comments by IDs
*IssueCommentsApi* | [**updateComment**](docs/IssueCommentsApi.md#updateComment) | **PUT** /rest/api/3/issue/{issueIdOrKey}/comment/{id} | Update comment
*IssueCustomFieldContextsApi* | [**addIssueTypesToContext**](docs/IssueCustomFieldContextsApi.md#addIssueTypesToContext) | **PUT** /rest/api/3/field/{fieldId}/context/{contextId}/issuetype | Add issue types to context
*IssueCustomFieldContextsApi* | [**assignProjectsToCustomFieldContext**](docs/IssueCustomFieldContextsApi.md#assignProjectsToCustomFieldContext) | **PUT** /rest/api/3/field/{fieldId}/context/{contextId}/project | Assign custom field context to projects
*IssueCustomFieldContextsApi* | [**createCustomFieldContext**](docs/IssueCustomFieldContextsApi.md#createCustomFieldContext) | **POST** /rest/api/3/field/{fieldId}/context | Create custom field context
*IssueCustomFieldContextsApi* | [**deleteCustomFieldContext**](docs/IssueCustomFieldContextsApi.md#deleteCustomFieldContext) | **DELETE** /rest/api/3/field/{fieldId}/context/{contextId} | Delete custom field context
*IssueCustomFieldContextsApi* | [**getContextsForField**](docs/IssueCustomFieldContextsApi.md#getContextsForField) | **GET** /rest/api/3/field/{fieldId}/context | Get custom field contexts
*IssueCustomFieldContextsApi* | [**getCustomFieldContextsForProjectsAndIssueTypes**](docs/IssueCustomFieldContextsApi.md#getCustomFieldContextsForProjectsAndIssueTypes) | **POST** /rest/api/3/field/{fieldId}/context/mapping | Get custom field contexts for projects and issue types
*IssueCustomFieldContextsApi* | [**getIssueTypeMappingsForContexts**](docs/IssueCustomFieldContextsApi.md#getIssueTypeMappingsForContexts) | **GET** /rest/api/3/field/{fieldId}/context/issueTypeMapping | Get issue types for custom field context
*IssueCustomFieldContextsApi* | [**removeCustomFieldContextFromProjects**](docs/IssueCustomFieldContextsApi.md#removeCustomFieldContextFromProjects) | **POST** /rest/api/3/field/{fieldId}/context/{contextId}/project/remove | Remove custom field context from projects
*IssueCustomFieldContextsApi* | [**updateCustomFieldContext**](docs/IssueCustomFieldContextsApi.md#updateCustomFieldContext) | **PUT** /rest/api/3/field/{fieldId}/context/{contextId} | Update custom field context

docs/Changelog.md


# Changelog

A changelog.
## Properties

Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**id** | **String** | The ID of the changelog. |  [optional] [readonly]
**author** | [**UserDetails**](UserDetails.md) | The user who made the change. |  [optional] [readonly]
**created** | **OffsetDateTime** | The date on which the change took place. |  [optional] [readonly]
**items** | [**List&lt;ChangeDetails&gt;**](ChangeDetails.md) | The list of items changed. |  [optional] [readonly]
**historyMetadata** | [**HistoryMetadata**](HistoryMetadata.md) | The history metadata associated with the changed. |  [optional] [readonly]





src/main/java/com/atlassian/jira/rest/client/model/User.java
/*
 * The Jira Cloud platform REST API
 * Jira Cloud platform REST API documentation
 *
 * The version of the OpenAPI document: 1001.0.0-SNAPSHOT
 * Contact: ecosystem@atlassian.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package com.atlassian.jira.rest.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.atlassian.jira.rest.client.model.AvatarUrlsBean;
import com.atlassian.jira.rest.client.model.SimpleListWrapperApplicationRole;
import com.atlassian.jira.rest.client.model.SimpleListWrapperGroupName;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonTypeName;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.net.URI;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

/**
 * A user with details as permitted by the user&#39;s Atlassian Account privacy settings. However, be aware of these exceptions:   *  User record deleted from Atlassian: This occurs as the result of a right to be forgotten request. In this case, &#x60;displayName&#x60; provides an indication and other parameters have default values or are blank (for example, email is blank).  *  User record corrupted: This occurs as a results of events such as a server import and can only happen to deleted users. In this case, &#x60;accountId&#x60; returns *unknown* and all other parameters have fallback values.  *  User record unavailable: This usually occurs due to an internal service outage. In this case, all parameters have fallback values.
 */
@ApiModel(description = "A user with details as permitted by the user's Atlassian Account privacy settings. However, be aware of these exceptions:   *  User record deleted from Atlassian: This occurs as the result of a right to be forgotten request. In this case, `displayName` provides an indication and other parameters have default values or are blank (for example, email is blank).  *  User record corrupted: This occurs as a results of events such as a server import and can only happen to deleted users. In this case, `accountId` returns *unknown* and all other parameters have fallback values.  *  User record unavailable: This usually occurs due to an internal service outage. In this case, all parameters have fallback values.")
@JsonPropertyOrder({
  User.JSON_PROPERTY_SELF,
  User.JSON_PROPERTY_KEY,
  User.JSON_PROPERTY_ACCOUNT_ID,
  User.JSON_PROPERTY_ACCOUNT_TYPE,
  User.JSON_PROPERTY_NAME,
  User.JSON_PROPERTY_EMAIL_ADDRESS,
  User.JSON_PROPERTY_AVATAR_URLS,
  User.JSON_PROPERTY_DISPLAY_NAME,
  User.JSON_PROPERTY_ACTIVE,
  User.JSON_PROPERTY_TIME_ZONE,
  User.JSON_PROPERTY_LOCALE,
  User.JSON_PROPERTY_GROUPS,
  User.JSON_PROPERTY_APPLICATION_ROLES,
  User.JSON_PROPERTY_EXPAND
})
@JsonTypeName("User")
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen", date = "2020-12-31T17:44:39.790417-08:00[America/Los_Angeles]")
public class User {
  public static final String JSON_PROPERTY_SELF = "self";
  private URI self;

  public static final String JSON_PROPERTY_KEY = "key";
  private String key;

  public static final String JSON_PROPERTY_ACCOUNT_ID = "accountId";
  private String accountId;

  /**
   * The user account type. Can take the following values:   *  &#x60;atlassian&#x60; regular Atlassian user account  *  &#x60;app&#x60; system account used for Connect applications and OAuth to represent external systems  *  &#x60;customer&#x60; Jira Service Desk account representing an external service desk
   */
  public enum AccountTypeEnum {
    ATLASSIAN("atlassian"),
    
    APP("app"),
    
    CUSTOMER("customer"),
    
    UNKNOWN("unknown");

    private String value;

    AccountTypeEnum(String value) {
      this.value = value;
    }

    @JsonValue
    public String getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    @JsonCreator
    public static AccountTypeEnum fromValue(String value) {
      for (AccountTypeEnum b : AccountTypeEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }
  }

  public static final String JSON_PROPERTY_ACCOUNT_TYPE = "accountType";
  private AccountTypeEnum accountType;

  public static final String JSON_PROPERTY_NAME = "name";
  private String name;

  public static final String JSON_PROPERTY_EMAIL_ADDRESS = "emailAddress";
  private String emailAddress;

  public static final String JSON_PROPERTY_AVATAR_URLS = "avatarUrls";
  private AvatarUrlsBean avatarUrls = null;

  public static final String JSON_PROPERTY_DISPLAY_NAME = "displayName";
  private String displayName;

  public static final String JSON_PROPERTY_ACTIVE = "active";
  private Boolean active;

  public static final String JSON_PROPERTY_TIME_ZONE = "timeZone";
  private String timeZone;

  public static final String JSON_PROPERTY_LOCALE = "locale";
  private String locale;

  public static final String JSON_PROPERTY_GROUPS = "groups";
  private SimpleListWrapperGroupName groups = null;

  public static final String JSON_PROPERTY_APPLICATION_ROLES = "applicationRoles";
  private SimpleListWrapperApplicationRole applicationRoles = null;

  public static final String JSON_PROPERTY_EXPAND = "expand";
  private String expand;


   /**
   * The URL of the user.
   * @return self
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The URL of the user.")
  @JsonProperty(JSON_PROPERTY_SELF)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public URI getSelf() {
    return self;
  }




  public User key(String key) {
    
    this.key = key;
    return this;
  }

   /**
   * This property is no longer available and will be removed from the documentation soon. See the [deprecation notice](https://developer.atlassian.com/cloud/jira/platform/deprecation-notice-user-privacy-api-migration-guide/) for details.
   * @return key
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "This property is no longer available and will be removed from the documentation soon. See the [deprecation notice](https://developer.atlassian.com/cloud/jira/platform/deprecation-notice-user-privacy-api-migration-guide/) for details.")
  @JsonProperty(JSON_PROPERTY_KEY)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getKey() {
    return key;
  }


  public void setKey(String key) {
    this.key = key;
  }


  public User accountId(String accountId) {
    
    this.accountId = accountId;
    return this;
  }

   /**
   * The account ID of the user, which uniquely identifies the user across all Atlassian products. For example, *5b10ac8d82e05b22cc7d4ef5*. Required in requests.
   * @return accountId
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The account ID of the user, which uniquely identifies the user across all Atlassian products. For example, *5b10ac8d82e05b22cc7d4ef5*. Required in requests.")
  @JsonProperty(JSON_PROPERTY_ACCOUNT_ID)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getAccountId() {
    return accountId;
  }


  public void setAccountId(String accountId) {
    this.accountId = accountId;
  }


   /**
   * The user account type. Can take the following values:   *  &#x60;atlassian&#x60; regular Atlassian user account  *  &#x60;app&#x60; system account used for Connect applications and OAuth to represent external systems  *  &#x60;customer&#x60; Jira Service Desk account representing an external service desk
   * @return accountType
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The user account type. Can take the following values:   *  `atlassian` regular Atlassian user account  *  `app` system account used for Connect applications and OAuth to represent external systems  *  `customer` Jira Service Desk account representing an external service desk")
  @JsonProperty(JSON_PROPERTY_ACCOUNT_TYPE)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public AccountTypeEnum getAccountType() {
    return accountType;
  }




  public User name(String name) {
    
    this.name = name;
    return this;
  }

   /**
   * This property is no longer available and will be removed from the documentation soon. See the [deprecation notice](https://developer.atlassian.com/cloud/jira/platform/deprecation-notice-user-privacy-api-migration-guide/) for details.
   * @return name
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "This property is no longer available and will be removed from the documentation soon. See the [deprecation notice](https://developer.atlassian.com/cloud/jira/platform/deprecation-notice-user-privacy-api-migration-guide/) for details.")
  @JsonProperty(JSON_PROPERTY_NAME)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getName() {
    return name;
  }


  public void setName(String name) {
    this.name = name;
  }


   /**
   * The email address of the user. Depending on the user’s privacy setting, this may be returned as null.
   * @return emailAddress
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The email address of the user. Depending on the user’s privacy setting, this may be returned as null.")
  @JsonProperty(JSON_PROPERTY_EMAIL_ADDRESS)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getEmailAddress() {
    return emailAddress;
  }




   /**
   * The avatars of the user.
   * @return avatarUrls
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The avatars of the user.")
  @JsonProperty(JSON_PROPERTY_AVATAR_URLS)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public AvatarUrlsBean getAvatarUrls() {
    return avatarUrls;
  }




   /**
   * The display name of the user. Depending on the user’s privacy setting, this may return an alternative value.
   * @return displayName
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The display name of the user. Depending on the user’s privacy setting, this may return an alternative value.")
  @JsonProperty(JSON_PROPERTY_DISPLAY_NAME)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getDisplayName() {
    return displayName;
  }




   /**
   * Whether the user is active.
   * @return active
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "Whether the user is active.")
  @JsonProperty(JSON_PROPERTY_ACTIVE)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public Boolean getActive() {
    return active;
  }




   /**
   * The time zone specified in the user&#39;s profile. Depending on the user’s privacy setting, this may be returned as null.
   * @return timeZone
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The time zone specified in the user's profile. Depending on the user’s privacy setting, this may be returned as null.")
  @JsonProperty(JSON_PROPERTY_TIME_ZONE)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getTimeZone() {
    return timeZone;
  }




   /**
   * The locale of the user. Depending on the user’s privacy setting, this may be returned as null.
   * @return locale
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The locale of the user. Depending on the user’s privacy setting, this may be returned as null.")
  @JsonProperty(JSON_PROPERTY_LOCALE)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getLocale() {
    return locale;
  }




   /**
   * The groups that the user belongs to.
   * @return groups
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The groups that the user belongs to.")
  @JsonProperty(JSON_PROPERTY_GROUPS)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public SimpleListWrapperGroupName getGroups() {
    return groups;
  }




   /**
   * The application roles the user is assigned to.
   * @return applicationRoles
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "The application roles the user is assigned to.")
  @JsonProperty(JSON_PROPERTY_APPLICATION_ROLES)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public SimpleListWrapperApplicationRole getApplicationRoles() {
    return applicationRoles;
  }




   /**
   * Expand options that include additional user details in the response.
   * @return expand
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "Expand options that include additional user details in the response.")
  @JsonProperty(JSON_PROPERTY_EXPAND)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getExpand() {
    return expand;
  }




  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    User user = (User) o;
    return Objects.equals(this.self, user.self) &&
        Objects.equals(this.key, user.key) &&
        Objects.equals(this.accountId, user.accountId) &&
        Objects.equals(this.accountType, user.accountType) &&
        Objects.equals(this.name, user.name) &&
        Objects.equals(this.emailAddress, user.emailAddress) &&
        Objects.equals(this.avatarUrls, user.avatarUrls) &&
        Objects.equals(this.displayName, user.displayName) &&
        Objects.equals(this.active, user.active) &&
        Objects.equals(this.timeZone, user.timeZone) &&
        Objects.equals(this.locale, user.locale) &&
        Objects.equals(this.groups, user.groups) &&
        Objects.equals(this.applicationRoles, user.applicationRoles) &&
        Objects.equals(this.expand, user.expand);
  }

  @Override
  public int hashCode() {
    return Objects.hash(self, key, accountId, accountType, name, emailAddress, avatarUrls, displayName, active, timeZone, locale, groups, applicationRoles, expand);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class User {\n");
    sb.append("    self: ").append(toIndentedString(self)).append("\n");
    sb.append("    key: ").append(toIndentedString(key)).append("\n");
    sb.append("    accountId: ").append(toIndentedString(accountId)).append("\n");
    sb.append("    accountType: ").append(toIndentedString(accountType)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    emailAddress: ").append(toIndentedString(emailAddress)).append("\n");
    sb.append("    avatarUrls: ").append(toIndentedString(avatarUrls)).append("\n");
    sb.append("    displayName: ").append(toIndentedString(displayName)).append("\n");
    sb.append("    active: ").append(toIndentedString(active)).append("\n");
    sb.append("    timeZone: ").append(toIndentedString(timeZone)).append("\n");
    sb.append("    locale: ").append(toIndentedString(locale)).append("\n");
    sb.append("    groups: ").append(toIndentedString(groups)).append("\n");
    sb.append("    applicationRoles: ").append(toIndentedString(applicationRoles)).append("\n");
    sb.append("    expand: ").append(toIndentedString(expand)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

